### PR TITLE
chore: add cloud distro CVE-2023-3955 and CVE-2023-3676

### PR DESCRIPTION
**Pull request by chen-keinan**
chore: add cloud distro CVE-2023-3955 and CVE-2023-3676

### CODE REVIEW

File: upstream/CVE-2023-3955.json [hunk 1]
```diff
@@ -53,12 +53,199 @@
...
+            ],
+            "ranges": [
+                {
+                    "type": "SEMVER",
+                    "events": [
+                        {
+                            "introduced": "1.27.0"
+                        },
+                        {
+                            "fixed": "1.27.4-eks-8ccc7ba"
```
Comment (knqyf263) [inline upstream/CVE-2023-3955.json:76]: I'm wondering if this version is comparable. 8ccc7ba looks like a commit hash.

File: upstream/CVE-2023-3955.json [hunk 2]
```diff
@@ -53,12 +53,199 @@
...
+            ],
+            "ranges": [
+                {
+                    "type": "SEMVER",
+                    "events": [
+                        {
+                            "introduced": "1.27.0"
+                        },
+                        {
+                            "fixed": "1.27.5-gke.200"
```
Comment (knqyf263) [inline upstream/CVE-2023-3955.json:119]: Did you make sure the kubelet version matches the cluster version?

File: upstream/CVE-2023-3955.json [hunk 3]
```diff
@@ -53,12 +53,199 @@
...
+                        },
+                        {
+                            "fixed": "1.24.15-hotfix.20230728"
+                        },
+                        {
+                            "introduced": "1.24.0"
+                        },
+                        {
+                            "fixed": "1.24.10-hotfix.20230728"
+                        }
```
Comment (knqyf263) [inline upstream/CVE-2023-3955.json:223]: These ranges are overwrapping. Which is correct?

**Review by knqyf263 — COMMENTED** [review]
My previous comments have gone. I left almost the same comments.

**Comment by chen-keinan** [comment]
> My previous comments have gone. I left almost the same comments.

I have to create a new PR as prev. PR was closed by history re-write made by LICENSE removal

File: upstream/CVE-2023-3955.json (hunk 1 above)
Comment (chen-keinan) [reply upstream/CVE-2023-3955.json:76]: yes its comparable but it is identified  `eks-8ccc7ba` as pre-release 

`1.27.4-eks-8ccc7ba  < 1.27.0`

Comment (knqyf263) [reply upstream/CVE-2023-3955.json:76]: It is not our desired result, right?

File: upstream/CVE-2023-3955.json (hunk 2 above)
Comment (knqyf263) [reply upstream/CVE-2023-3955.json:119]: If I understand correctly, 1.24.17-gke.200 is a fixed version for GKE, not kubelet. 

>we recommend that you [manually upgrade](https://cloud.google.com/kubernetes-engine/docs/how-to/upgrading-a-cluster) your cluster and node pools to one of the following GKE versions:
>
>1.24.17-gke.200
>1.25.13-gke.200
>1.26.8-gke.200
>1.27.5-gke.200
>1.28.1-gke.200

I'm wondering if kubelet also has this versioning.

Comment (chen-keinan) [reply upstream/CVE-2023-3955.json:119]: in cloud kubernetes services a user cannot bump kubelet by it self. the bump is on `Node poll` where kubelet live.
the vulnerability is in kubelet and the fix is also.

GKE (same as kubernetes) version is the same as kubelet version.

Comment (knqyf263) [reply upstream/CVE-2023-3955.json:119]: >GKE (same as kubernetes) version is the same as kubelet version.

Did you ensure it? I'm just concerned our KBOM detects the kubelet version, but the versioning is different from `1.y.z-gke.xxx`. It results in false detection.

File: upstream/CVE-2023-3955.json (hunk 1 above)
Comment (chen-keinan) [reply upstream/CVE-2023-3955.json:76]: true, we need to figure out how to solve this use case

File: upstream/CVE-2023-3955.json (hunk 2 above)
Comment (chen-keinan) [reply upstream/CVE-2023-3955.json:119]: yes , all core component versions are the same a kubernetes release version

**Pull request closed without merging**